tools/osbuild-depsolve-dnf5

```diff
@@ -0,0 +1,500 @@
+    'x86_64': ('x86_64', 'amd64', 'ia32e'),
+})
+
+
+def remote_location(package, schemes=("http", "ftp", "file", "https")):
+    """Return the remote url where a package rpm may be downloaded from
+
+    This wraps the get_remote_location() function, returning the first
+    result or if it cannot find a suitable url it raises a RuntimeError
+    """
+    urls = package.get_remote_locations(schemes)
+    if not urls or len(urls) == 0:
+        raise RuntimeError(f"Cannot determine remote location for {package.get_nevra()}")
+
+    return urls[0]
+
+
+class TransactionError(Exception):
+    pass
+
+
+class RepoError(Exception):
+    pass
+
+
+class Solver():
+    """Solver implements package related actions
+
+    These include depsolving a package set, searching for packages, and dumping a list
+    of all available packages.
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, repos, module_platform_id, persistdir, cachedir, arch, exclude_pkgs=None):
+        if not exclude_pkgs:
+            exclude_pkgs = []
+
+        # substitutions to use with the repo urls
+        self.substitutions = {
+            "arch": arch,
+            "basearch": _BASEARCH_MAP[arch]
+        }
+        self.base = dnf5.base.Base()
+        self.base.get_vars().set("arch", arch)
+
+        # Enable fastestmirror to ensure we choose the fastest mirrors for
+        # downloading metadata (when depsolving) and downloading packages.
+        conf = self.base.get_config()
+        conf.fastestmirror = True
+
+        # Weak dependencies are installed for the 1st transaction
+        # This is set to False for any subsequent ones in depsolve()
+        conf.install_weak_deps = True
+
+        # We use the same cachedir for multiple architectures. Unfortunately,
+        # this is something that doesn't work well in certain situations
+        # with zchunk:
+        # Imagine that we already have cache for arch1. Then, we use dnf-json
+        # to depsolve for arch2. If ZChunk is enabled and available (that's
+        # the case for Fedora), dnf will try to download only differences
+        # between arch1 and arch2 metadata. But, as these are completely
+        # different, dnf must basically redownload everything.
+        # For downloding deltas, zchunk uses HTTP range requests. Unfortunately,
+        # if the mirror doesn't support multi range requests, then zchunk will
+        # download one small segment per a request. Because we need to update
+        # the whole metadata (10s of MB), this can be extremely slow in some cases.
+        # I think that we can come up with a better fix but let's just disable
+        # zchunk for now. As we are already downloading a lot of data when
+        # building images, I don't care if we download even more.
+        conf.zchunk = False
+
+        # Set the rest of the dnf configuration.
+        conf.module_platform_id = module_platform_id
+        conf.config_file_path = "/dev/null"
```

Comment:
(super nitpick)  I wonder if we could/should report a bugreport that this should be support something like `None` or `""` to unset a config_file_path, it irks me a little bit that this is /dev/null - but feel free to ignore, it's just me being pedantic.

Reply:
This defaults to `/etc/dnf/dnf.conf` which we don't want it to read, we want to setup all of the configuration ourselves. It might be possible to convince them to change things, but since it's already reading a file it's easier to just pass `/dev/null`.